src/energy_systems/heat_sources/geothermal_probes.jl

```diff
@@ -1,315 +1,862 @@
+
+        borehole_thermal_resistance = R_1 / (2 * unit.probe_type)
+    elseif unit.model_type == "simplified"
+        # constant borehole thermal resistance from user input
+        borehole_thermal_resistance = unit.borehole_thermal_resistance
+    else
+        @error "Undefined model type $(unit.model_type) of unit $(unit.uac). Has to be one of: 'simplified', 'detailed'."
+        exit()
+    end
+
+    # calculate new average fluid temperature with g-function approach
+    if unit.time_index == 1
+        unit.power_in_out_difference_per_probe_meter[unit.time_index] = wh_to_watts(unit.energy_in_out_per_probe_meter[unit.time_index])
     else
-        # timestep has not been calculated yet (normal case)
-        unit.temperature_field = unit.temperature_field .+ energy/1000 - (unit.temperature_field .- unit.ambient_temperature)./200
+        unit.power_in_out_difference_per_probe_meter[unit.time_index] = wh_to_watts(unit.energy_in_out_per_probe_meter[unit.time_index] - unit.energy_in_out_per_probe_meter[unit.time_index-1])
     end
 
-    # update information if the current timestep has already been calculated
-    unit.last_timestep_calculated = current_timestep
+    current_temperature_difference = sum(reverse(unit.power_in_out_difference_per_probe_meter[1:unit.time_index]) .* unit.g_function[1:unit.time_index]) / (2 * pi * unit.soil_heat_conductivity)
 
+    borehole_current_wall_temperature = unit.soil_undisturbed_ground_temperature + current_temperature_difference
+
+    fluid_temperature = borehole_current_wall_temperature + wh_to_watts(unit.energy_in_out_per_probe_meter[unit.time_index]) * borehole_thermal_resistance
+    return fluid_temperature
 end
 
-# process function that provides energy from the geothermal probes and calculates new temperatures 
+"""
+    calculate_alpha_pipe(unit::GeothermalProbes)
+
+Calculates the convective heat transfer coefficient alpha in the pipe of the probe.
+Uses parameters stored in unit::GeothermalProbes to calculate the mass flow in the pipe,
+the Reynolds number, the Nusselt number for laminar or turbulent flow and finally alpha. 
+
+Args:
+    unit::GeothermalProbes:         the unit struct of the geothermal probe with all its parameters
+Returns:
+    alpha::Float64:                 convective heat transfer coefficient in pipe in current time step
+    fluid_reynolds_number::Float64  the current Reynolds number of the fluid in the pipe
+"""
+function calculate_alpha_pipe(unit::GeothermalProbes)
+    # calculate mass flow in pipe
+    power_in_out_per_pipe = wh_to_watts(abs(unit.energy_in_out_per_probe_meter[unit.time_index])) * unit.probe_depth / unit.probe_type  # W/pipe
+    temperature_spread = unit.energy_in_out_per_probe_meter[unit.time_index] > 0 ? unit.loading_temperature_spread : unit.unloading_temperature_spread
+    mass_flow_per_pipe = power_in_out_per_pipe / (unit.fluid_specific_heat_capacity * temperature_spread)  # kg/s
+
+    use_dynamic_fluid_properties = false
```

Comment:
Is this supposed to be customizable? Or a feature that's not working yet?

Reply:
I have remove the flag and commented out the dynamic fluid parameters for potential further improvements of the model.